Lib/posixpath.py

```diff
@@ -251,12 +236,9 @@ def expanduser(path):
         if 'HOME' not in os.environ:
             try:
                 import pwd
-            except ImportError:
-                # pwd module unavailable, return path unchanged
-                return path
-            try:
                 userhome = pwd.getpwuid(os.getuid()).pw_dir
-            except KeyError:
+            except (ImportError, KeyError):
```

Comment:
Apologies if I'm beginning to sound like a broken record, but does this change improve performance? If not, could you please undo it?

Reply:
Not to sound rude, but AlexWayGood already pointed this out: https://github.com/python/cpython/pull/117350#discussion_r1545265582
I'm still looking into it, don't worry (I'll close his conversation as it does no longer show up in the files changed).

Reply:
> Previously it was clear for a reader of the code exactly which statement might raise `ImportError` and exactly which statement might raise `KeyError`. By grouping them together into a single `try`/`except` block, we lose that clarity.
> 
> Does grouping them together speed things up here? How much by?

Reply:
It doesn't seem to speed things up, reverted.